grading/execute.cpp

```diff
@@ -70,13 +61,13 @@ bool system_program(const std::string &program, std::string &full_path_executabl
         allowed_system_programs["php"] = "/usr/bin/php";
     }
     if (allowed_system_programs.contains(program)) {
-        full_path_executable = replace_placeholder(allowed_system_programs[program]);
+        full_path_executable = allowed_system_programs[program];
         return true;
     }
 
     for (auto& x : allowed_system_programs.items())
```

Comment:
Just one more comment considering that this program should be as performant as possible. This loop is encountered only if `allowed_system_programs.contains(program)` returns false, would it be a good idea to wrap this/these loop(s) in a conditional such that if `program` contains `/` then we can directly skip to this loop instead of first running `allowed_system_programs.contains(program)`? Because I don't think if `/` is present in `program` then it will be one of the keys in JSON. Thoughts?

Reply:
If I understand correctly you are suggesting `if (program.length() > 0 && program[0] == "/") ` check right

Reply:
Right, something like:
`if (program.length() > 0 && program[0] == "/") {
    for (auto& x : allowed_system_programs.items())
    {
        if (x.value() == program) {
           full_path_executable = program;
           return true;
        }
    }
}
else {
    if (allowed_system_programs.contains(program)) {
        full_path_executable = allowed_system_programs[program];
        return true;
    }
}`

Reply:
Updated